File: java/eyes.sdk.core/src/main/java/com/applitools/eyes/fluent/SimpleRegionByRectangle.java
package com.applitools.eyes.fluent;

import com.applitools.eyes.Padding;
import com.applitools.eyes.Region;
import com.fasterxml.jackson.annotation.JsonProperty;

public class SimpleRegionByRectangle implements GetRegion {
    private final Region region;
    private final Padding padding;

    public SimpleRegionByRectangle(Region region) {
        this.region = region;
        this.padding = new Padding();
    }

    public SimpleRegionByRectangle(Region region, Padding padding) {
        this.region = region;
        this.padding = padding;
    }

    @JsonProperty("region")
    public Region getRegion() {
        return region;
    }

    public Padding getPadding() { return this.padding; }
}

File: dotnet/Eyes.Images.DotNet/Fluent/SimpleRegionByRectangle.cs
using System.Drawing;
using Applitools.Fluent;
using Applitools.Utils.Geometry;

namespace Applitools
{
    public class SimpleRegionByRectangle : IGetRegions
    {
        private readonly Padding padding_;
        private readonly string regionId_;
        private readonly IMutableRegion region_;

        public SimpleRegionByRectangle(Rectangle region, Padding padding = null, string regionId = null)
        {
            padding_ = padding;
            regionId_ = regionId;
            region_ = new MutableRegion(region);
        }

        public SimpleRegionByRectangle(Point location, Size size) : this(new Rectangle(location, size)) { }
        
        public CodedRegionReference ToRegion()
        {
            return new CodedRegionReference
            {
                Region = new RegionRectangle
                {
                    X = region_.Left,
                    Y = region_.Top,
                    Width = region_.Width,
                    Height = region_.Height
                },
                Padding = padding_,
                RegionId = regionId_
            };
        }
    }
}